---
title: "rROMA"
author: "Luca Albergante (luca.albergante@gmail.com)"
date: "`r Sys.Date()`"
output:
  md_document:
    variant: markdown_github
---

The rROMA package is being developed to simplify the use of [ROMA](https://github.com/sysbio-curie/Roma) from R. It is currently under development.

## Intalling rROMA

The package is currently under development, non very well documented, and only available on GitHub. A functional Java Virtual machine for your system is necessary. Before installing the package, it is advisable to install rJava from sources by using

```{r, eval=FALSE}
install.packages(pkgs = "rJava", repos="http://rforge.net", type = 'source')
```

Compiling from source requires the appropriate development tools, e.g., C/C++ compiler. The installation of the package requires the devtools package, which is available from CRAN. The rpgraph package can be installed using

```{r, eval=FALSE}
install.packages("devtools")
library(devtools)
install_github("Albluca/rROMA")
```

## Using rROMA

rROMA exports the single function `rRoma` which provides a subset of the functionalites of ROMA. The function requires a matrix with gene expression data across multiple samples and a module list. In the following example artificial gene expression matrix and module list will be created and ROMA is executed. Just as the ROMA executable, only `ExpressionMatrix` and `ModuleList` are mandatory. All of the parameter of the function have the same function of the Java executable. Parameters that are not present in the argument list have not been implemented yet.

```{r, eval=TRUE}
library(rRoma)

nGenes <- 300
nSamples <- 100
nModules <- 5
GenesPerModule <- 100

GeneNames <- paste("Gene_", 1:nGenes, sep='')
SampleNames <- paste("Sample_", 1:nSamples, sep='')

NumericalExpressionMatrix <- matrix(rlnorm(n = nGenes*nSamples), nrow = nGenes, nSamples)
ExpressionMatrix <- cbind(GeneNames, NumericalExpressionMatrix)
colnames(ExpressionMatrix) <- c("Gene", SampleNames)

ModuleList <- list()

for(i in 1:nModules){
  ModuleList[[i]] <- list(Name=paste("Module", i), Desc="Descripion", Genes=sample(GeneNames, GenesPerModule))
}


Tables <- rRoma(ExpressionMatrix = ExpressionMatrix, ModuleList = ModuleList, fillMissingValues = 0, centerData = TRUE,
                doubleCenterData = FALSE, outlierThreshold = 2, typeOfPCAUsage = 0, robustPCAcalculation = TRUE,
                robustPCAcalculationForSampling = TRUE, numberOfGeneSetSizesToSample = 5, mostContributingGenesZthreshold = 1,
                diffSpotGenesZthreshold = 1, correlationThreshold = 0.6, graphicalOutputThreshold = 0.05,
                minimalNumberOfGenesInModule = 10, maximalNumberOfGenesInModule = 1000, minimalNumberOfGenesInModuleFound = 8,
                numberOfPermutations = 0, typeOfModuleFile = 0, saveDecomposedFiles = TRUE)

```

After running the function, `Tables` will be a list containing the tables that the Java command would have produced, with the only exclusion of numerical ".dat" files, which are ignored. The names of the elements of the list indicates the name of the file produced by the Java executable. For example the module scores can be obtained by typing

```{r}
Tables$module_scores.xls
```


Note that `ModuleList` must possess exactly three fields: `Name`, which is the name of the module, `Desc`, which is a description string, and `Genes`, which is an array of genes. Gene weights can be specified as in GMT files by appending the value encloded in a square parenthesis to the gene name, e.g., "Gene_1[+5]".

`ExpressionMatrix` must be a numeric, with missing values encoded by `NA`. The first column must indicate the gene namas and the names of the columns must indicate the names of the samples (except for the first one, which marks the gene name column).

